lib/compress.js

```diff
@@ -338,7 +338,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant(compressor)) {
+                    if (!(var_decl.value instanceof AST_RegExp) && var_decl.value.is_constant(compressor)) {
```

Comment:
I'm aware of that. This is getting to be very off topic to this ticket, but here's some background: 

The original code for the `x?1:1 --> 1` `conditionals` optimization only checked for AST_Constants which includes AST_RegExp. That check was replaced with `Node.is_constant()` which is functionally the same as an `instanceof AST_Constant` check except it recognizes the very common `!0` style expressions as well. The original author of the `conditionals` optimization probably did not want to waste CPU cycles comparing the ASTs of the consequent and alternative of every ternary operator if equivalence is less likely to occur. AST comparison is expensive and involves calling `AST_Node.print_to_string()` on both expressions and comparing the resultant strings.

In any case, this PR works correctly and there are no regressions in other parts of the code. If you have issue with other optimizations please open a new ticket.